app/controllers/playspaces_controller.rb

```diff
@@ -38,4 +37,49 @@ def playspaces_params
     params.require(:playspace).permit(:name, :address, :district, :region, :bike_path, :merry_go_round, :play_structure, :seesaw, :slide, :swing, :carpark, :snack_shop, :wc, :category_id)
   end
 
+  def filter_params
+    params.require(:playspace).permit(:name, :district, :region, :category_id, :bike_path, :merry_go_round, :play_structure, :seesaw, :slide, :swing, :carpark, :snack_shop, :wc)
+  end
+
+  def filter_queries(results=nil)
+    require 'wannabe_bool'
+    results = results || Playspace.all
+
+    if params[:playspace] != nil
+      puts "started Filtering"
+      f_params = filter_params
+      name = f_params[:name] == nil || f_params[:name] == "" ? nil : f_params[:name].downcase
```

Comment:
$(document).ready(function(){
  var option = {
    HK: ["Central", "North Point", "Repulse Bay", "The Peak", "Wan Chai", "Western"],
    Kln: ["Ho Man Tin", "Tsim Sha Tsui", "Wong Tai Sin"],
    NT: ["Sai Kung", "Sha Tin", "Tai Po"]
  }
  
var bindRegionDistrict = function (formID) {

  $(formID + " #playspace_region").on('change', function() {
    $(formID + " #playspace_district").html("");
    $(formID + " #playspace_district").append("<option value>Select by district:</option>")


    option.each |r|
      for (var i = 0; i < option[r].length; i++) {
        $(formID + " #playspace_district").append("<option value='"+option[r][i]+"'>"+option[r][i]+"</option>")
      }
    }

  })
};

  bindRegionDistrict('#search')
  bindRegionDistrict('#new_playspace')

})